### PR TITLE
Detect package requirements for plugins.

### LINKED ISSUE #322: Feature: Plugin Dependencies
## Feature Details
Sort out how to make sure dependencies included in plugins (but not in {gsm.app}) are noticed during deployment.

## Additional Comments
It's POSSIBLE this won't be a problem at all in normal usage. The only case I can think of for certain right now is the contrived one I'm running into: I have an example plugin inside gsm.app that has dependencies that aren't direct gsm.app dependencies (or wouldn't be if I didn't explicitly add them). Work through real use-cases and see if anything would be hidden.

### LINKED ISSUE #323: QC: Plugins Vignette
## QC Details
Write a vignette about how to develop and use plugins.

### LINKED ISSUE #339: Bugfix: KRIs with Flags Break in Analytics Output
## Expected Behavior
See https://github.com/Gilead-BioStats/gsm/issues/1983 for most of the info. 

If a site doesn't have any flags, display a placeholder GT table. This isn't perfect but it's an easy, stop-gap fix.

To test, update app.R, adding this filter to dfResults: `dfResults = dplyr::filter(gsm.app::sample_dfResults, GroupID != "0X159")`

(I fixed it before writing this ticket).

### PR DESCRIPTION
**Pull request by jonthegeek**
## Overview
Be better about detecting packages in plugin specs.

This is a work in progress, but I want to make sure it deploys properly at this point.

Right now this is fairly manual. I'm working toward automating it more. We'll make the user tell us SOME things, but I'm trying to make it a light lift.

## Connected Issues

- Closes #322 
- Closes #323 
- Closes #339

### CODE REVIEW

**Comment by jonthegeek** [comment]
The latest commit addresses https://github.com/Gilead-BioStats/gsm/issues/1983

File: R/mod_Plugins_UI.R [hunk 1]
```diff
@@ -1,12 +1,14 @@
...
 #' @inheritParams shared-params
+#' @returns Plugins UI elements, either as a tab (if there is only one plugin)
+#'   or in a drop-down list (for multiple plugins).
 #' @keywords internal
 mod_Plugins_UI <- function(id, lPlugins = NULL) {
   ns <- NS(id)
   if (!is.null(lPlugins)) {
     plugin_items <- purrr::imap(lPlugins, function(lPlugin, i) {
-      fnUI <- rlang::as_function(lPlugin$shiny$UI)
+      fnUI <- util_AsFunction(lPlugin$shiny$UI)
```
Comment (lauramaxwell) [inline R/mod_Plugins_UI.R:11]: so, once we fix up issue 1992 over in gsm, you'll use that function here, or is there still additional functionality that this method provides?

**Review by lauramaxwell — APPROVED** [review]
This is great! I'm wondering if eventually this dependency-checking functionality should be done over in gsm.template, or if it makes sense for this to live here indefinitely. The centralization of requirement- and spec-gathering to either gsm or gsm.template is appealing to me, but it might make the web of gsm more complicated than it's worth. Anyways- this is good to go, and i'm sure we'll have related discussions about where this lives in the new year!

File: R/mod_Plugins_UI.R (hunk 1 above)
Comment (jonthegeek) [reply R/mod_Plugins_UI.R:11]: TECHNICALLY I also have functionality in here to deal with a situation where a user sends a function in (without using plugin_Read() to generate their lPlugins entry). I don't know if that's important for sure. If it is, we should move it over to the central function (which I call inside `util_AsFunction()` right now).

**Comment by jonthegeek** [comment]
> This is great! I'm wondering if eventually this dependency-checking functionality should be done over in gsm.template, or if it makes sense for this to live here indefinitely.
I think gsm.app should include (or wrap) any functionality that's needed to deploy an app, even if we also use it somewhere else. Putting it in gsm.core might make sense, and then gsm.app and gsm.template can both use it.

For some reason the shinyapps-deploy is taking forever this time. I want to make sure that still works before I merge this, but I should be able to merge after that, and then tomorrow I'll sort out a release!

**Comment by jonthegeek** [comment]
It looks like shinyapps uses some sort of cache of github or something. It couldn't see `"Gilead-BioStats/gsm@v2.2.0"`. I'm trying again, then will look closer to see if I need to bust the cache or something.

**Comment by jonthegeek** [comment]
Posit is working on a fix: https://forum.posit.co/t/unable-to-deploy-to-shinyapps-io-due-to-package-dependency/195372/9

Giving it "Private repo access" in my shinyapps.io profile would fix it... but I've already done that in my DSLC shinyapps account, and it doesn't let you link a GitHub user to more than one account. In theory I might be able to fix it via .Renviron or something similar, I'll see if I can make that work cleanly. I'm glad to at least have an answer that I'm not crazy/doing something wrong!

**Review by jonthegeek — COMMENTED** [review]
Try force-installing it again.

**Pull request merged**